src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -667,6 +668,40 @@ public override ImmutableArray<string> TupleElementNames
         }
 
         /// <summary>
+        /// Returns the symbols for all the tuple fields (in order and cached).
+        /// </summary>
+        public override ImmutableArray<FieldSymbol> TupleElementFields
+        {
+            get
+            {
+                if (_lazyFields.IsDefault)
+                {
+                    ImmutableInterlocked.InterlockedInitialize(ref _lazyFields, CreateFields());
+                }
+
+                return _lazyFields;
+            }
+        }
+
+        private ImmutableArray<FieldSymbol> CreateFields()
+        {
+            var builder = ArrayBuilder<FieldSymbol>.GetInstance(_elementTypes.Length);
+            var fields = GetMembers().OfType<FieldSymbol>();
```

Comment:
We prefer Kind checks to type checks. It is more efficient to inline it into the loop through all GetMembers()